### PR TITLE
Support for Postgres unitemporal delta milestoning with tests 

### PR DESCRIPTION
**Pull request by rengam32**
#### What type of PR is this?

Improvement

#### What does this PR do / why is it needed ?

- Support for Postgres unitemporal delta milestoning with tests 
- SQL tests for non-temporal snaphot milestoning 

#### Which issue(s) this PR fixes:
None

#### Does this PR introduce a user-facing change?
No

### CODE REVIEW

File: legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-core/src/main/java/org/finos/legend/engine/persistence/components/relational/api/RelationalIngestorAbstract.java [hunk 1]
```diff
@@ -191,7 +191,7 @@ public String getIngestRunId()
     public abstract RelationalSink relationalSink();
 
     @Derived
-    protected TransformOptions transformOptions()
+    public TransformOptions transformOptions()
```
Comment (kumuwu) [inline legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-core/src/main/java/org/finos/legend/engine/persistence/components/relational/api/RelationalIngestorAbstract.java:194]: i think can revert this back

File: legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-postgres/src/test/java/org/finos/legend/engine/persistence/components/e2e/TestUtils.java [hunk 2]
```diff
@@ -659,6 +659,14 @@ public static Dataset getCsvDatasetRefWithLessColumnsThanMainForBitemp(String da
             .build();
     }
 
+    public static Dataset getCsvDatasetRefWithLessColumnsThanMain(String dataPath)
```
Comment (kumuwu) [inline legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-postgres/src/test/java/org/finos/legend/engine/persistence/components/e2e/TestUtils.java:662]: Dont think we need this as we are not testing ExternalDatasetReference anymore

File: legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-postgres/src/test/resources/data/unitemporal-incremental-milestoning/input/batch_id_based/without_delete_ind/staging_data_pass1.json [hunk 3]
```diff
@@ -0,0 +1,26 @@
+[
```
Comment (kumuwu) [inline legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-postgres/src/test/resources/data/unitemporal-incremental-milestoning/input/batch_id_based/without_delete_ind/staging_data_pass1.json:1]: we dont need this file

File: legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-postgres/src/test/resources/data/unitemporal-incremental-milestoning/input/batch_id_based/without_delete_ind/staging_data_pass2.json [hunk 4]
```diff
@@ -0,0 +1,26 @@
+[
```
Comment (kumuwu) [inline legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-postgres/src/test/resources/data/unitemporal-incremental-milestoning/input/batch_id_based/without_delete_ind/staging_data_pass2.json:1]: we dont need this file

File: legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-postgres/src/test/resources/data/unitemporal-incremental-milestoning/input/batch_id_based/without_delete_ind/staging_data_pass3.json [hunk 5]
```diff
@@ -0,0 +1 @@
+[]
```
Comment (kumuwu) [inline legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-postgres/src/test/resources/data/unitemporal-incremental-milestoning/input/batch_id_based/without_delete_ind/staging_data_pass3.json:1]: we dont need this file

Comment (kumuwu) [reply legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-postgres/src/test/resources/data/unitemporal-incremental-milestoning/input/batch_id_based/without_delete_ind/staging_data_pass3.json:1]: Also it seems like the data files for bitemporal snapshot are included, we can delete them

File: legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-postgres/src/test/java/org/finos/legend/engine/persistence/components/e2e/unitemporal/UnitemporalDeltaWithBatchIdTest.java [hunk 6]
```diff
@@ -0,0 +1,522 @@
...
+    }
+
+    @Test
+    void testUniTemporalDeltaWithAllVersionGreaterThanAndStagingFilters() throws Exception
+    {
+        DatasetDefinition mainTable = TestUtils.getDefaultMainTable();
+        DatasetDefinition stagingDataset = DatasetDefinition.builder()
+                .group(testSchemaName)
+                .name(stagingTableName)
+                .schema(TestDedupAndVersioning.baseSchemaWithVersionAndBatch)
```
Comment (kumuwu) [inline legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-postgres/src/test/java/org/finos/legend/engine/persistence/components/e2e/unitemporal/UnitemporalDeltaWithBatchIdTest.java:343]: I have already added the schema in TestUtils previously, we can just call **getDedupAndVersioningSchemaWithVersionAndBatch** here, then we dont need to duplicate the whole TestDedupAndVersioning file here

File: legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-postgres/src/test/java/org/finos/legend/engine/persistence/components/e2e/unitemporal/UnitemporalDeltaWithBatchIdTest.java [hunk 7]
```diff
@@ -0,0 +1,522 @@
...
+
+        createStagingTableWithoutPks(stagingDataset);
+        DerivedDataset stagingTable = DerivedDataset.builder()
+                .group(testSchemaName)
+                .name(stagingTableName)
+                .schema(TestDedupAndVersioning.baseSchemaWithVersion)
+                .addDatasetFilters(DatasetFilter.of("batch", FilterType.EQUAL_TO, 1))
+                .build();
+        String path = basePathForInput + "with_all_version/data1.csv";
+        TestDedupAndVersioning.loadDataIntoStagingTableWithVersionAndBatch(path);
```
Comment (kumuwu) [inline legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-postgres/src/test/java/org/finos/legend/engine/persistence/components/e2e/unitemporal/UnitemporalDeltaWithBatchIdTest.java:354]: The corresponding load method is also already added in BaseTest, it is called **loadDedupAndVersioningStagingDataWithVersionAndBatch**

File: legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-postgres/src/test/java/org/finos/legend/engine/persistence/components/e2e/unitemporal/UnitemporalDeltaWithBatchIdTest.java [hunk 8]
```diff
@@ -0,0 +1,522 @@
...
+    }
+
+    @Test
+    void testUniTemporalDeltaWithAllVersionDigestBasedAndStagingFilters() throws Exception
+    {
+        DatasetDefinition mainTable = TestUtils.getDefaultMainTable();
+        DatasetDefinition stagingDataset = DatasetDefinition.builder()
+                .group(testSchemaName)
+                .name(stagingTableName)
+                .schema(TestDedupAndVersioning.baseSchemaWithVersionAndBatch)
```
Comment (kumuwu) [inline legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-postgres/src/test/java/org/finos/legend/engine/persistence/components/e2e/unitemporal/UnitemporalDeltaWithBatchIdTest.java:436]: Similar as above, we dont need TestDedupAndVersioning file

File: legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-postgres/src/test/java/org/finos/legend/engine/persistence/components/e2e/versioning/TestDedupAndVersioning.java [hunk 9]
```diff
@@ -0,0 +1,830 @@
+// Copyright 2024 Goldman Sachs
```
Comment (kumuwu) [inline legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-postgres/src/test/java/org/finos/legend/engine/persistence/components/e2e/versioning/TestDedupAndVersioning.java:1]: After changing the above methods to use the ones in TestUtils and BaseTest, we dont need to duplicate this file here, and in turn we can delete the data files too, to keep it cleaner

**Review by epsstan — APPROVED** [review]

**Pull request merged**